src/test/java/org/folio/verticles/CancelledJobExecutionConsumerVerticleTest.java

```diff
@@ -0,0 +1,158 @@
+package org.folio.verticles;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.folio.ApiTestSuite.KAFKA_ENV_VALUE;
+import static org.folio.ApiTestSuite.observeTopic;
+import static org.folio.ApiTestSuite.sendToTopic;
+import static org.folio.DataImportEventTypes.DI_COMPLETED;
+import static org.folio.DataImportEventTypes.DI_ERROR;
+import static org.folio.DataImportEventTypes.DI_INCOMING_EDIFACT_RECORD_PARSED;
+import static org.folio.DataImportEventTypes.DI_JOB_CANCELLED;
+import static org.folio.dataimport.handlers.events.DataImportKafkaHandler.JOB_PROFILE_SNAPSHOT_ID_KEY;
+import static org.folio.kafka.KafkaTopicNameHelper.getDefaultNameSpace;
+import static org.folio.rest.impl.MockServer.addMockEntry;
+import static org.folio.rest.jaxrs.model.EntityType.EDIFACT_INVOICE;
+import static org.folio.rest.jaxrs.model.EntityType.INVOICE;
+import static org.folio.rest.jaxrs.model.ProfileType.ACTION_PROFILE;
+import static org.folio.rest.jaxrs.model.ProfileType.JOB_PROFILE;
+import static org.folio.rest.jaxrs.model.ProfileType.MAPPING_PROFILE;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.junit5.VertxTestContext;
+import lombok.SneakyThrows;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.folio.ActionProfile;
+import org.folio.ApiTestSuite;
+import org.folio.DataImportEventPayload;
+import org.folio.JobProfile;
+import org.folio.MappingProfile;
+import org.folio.kafka.KafkaTopicNameHelper;
+import org.folio.processing.events.EventManager;
+import org.folio.processing.events.services.handler.EventHandler;
+import org.folio.rest.impl.ApiTestBase;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.rest.jaxrs.model.ProfileSnapshotWrapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class CancelledJobExecutionConsumerVerticleTest extends ApiTestBase {
```

Comment:
Need to add this test into ApiTestSuite, until we have some approach to pick up it automatically